docs/docs/end-to-end-testing.md

```diff
@@ -0,0 +1,50 @@
+---
+title: "End-to-end testing"
+---
+
+[Cypress](https://www.cypress.io/) is one of the options when it comes to end-to-end testing. Setting up and writing tests in Cypress is developer-friendly and easy to learn. You'll profit from the time travel feature or the extensive debugging capabilities with Chrome DevTools. Of course you can also use it with Gatsby and this guide will explain how.
+
+In order to run Gatsby's development server and Cypress at the same time you'll use the little helper [start-server-and-test](https://github.com/bahmutov/start-server-and-test). If you're already using [react-testing-library](docs/testing-react-components) you might want to install `cypress-testing-library`, too. Install the following packages to your `devDependencies`:
+
+```sh
+npm install --save-dev cypress start-server-and-test
+```
+
+We also want the urls used by `cy.visit()` or `cy.request()` to be prefixed hence you have to create the file `cypress.json` at the root of your project with the following content:
+
+```json
+{
+  "baseUrl": "http://localhost:8000/"
+}
+```
+
+Last but not least you add additional scripts to your `package.json` to run Cypress:
+
+```json
+{
+  "scripts": {
+    "develop": "gatsby develop",
+    "cy:open": "cypress open",
+    "test:e2e": "start-server-and-test develop http://localhost:8000 cy:open"
+  }
+}
+```
+
+Run `test:e2e` in your command line and see Cypress running for the first time. A folder named `cypress` will be created at the root of your project and a new application window will pop up. Cypress' [getting started guide](https://docs.cypress.io/guides/getting-started/writing-your-first-test.html#) is a good start to learn how to write tests!
```

Comment:
Make sure the link text includes [Cypress' getting started guide] so that no one things the link is to Gatsby's getting started guide.